### PR TITLE
Brace/anthropic tools

### PR DESCRIPTION
**Pull request by bracesproul**

### CODE REVIEW

File: libs/langchain-anthropic/src/chat_models.ts [hunk 1]
```diff
@@ -233,6 +343,35 @@ export class ChatAnthropicMessages<
...
+    const body = tools
+      ? {
+          ...request,
+          tools,
+        }
+      : undefined;
+    const headers = tools ? toolUseBetaHeader : undefined;
+    return {
+      signal: options.signal,
+      ...(body ? { body } : {}),
```
Comment (jacoblee93) [inline libs/langchain-anthropic/src/chat_models.ts:370]: This is confusing but I guess we'll delete it soon anyway

File: libs/langchain-anthropic/src/chat_models.ts [hunk 2]
```diff
@@ -233,6 +343,35 @@ export class ChatAnthropicMessages<
...
+      ...(headers ? { headers } : {}),
```
Comment (jacoblee93) [inline libs/langchain-anthropic/src/chat_models.ts:371]: Same

File: libs/langchain-anthropic/src/chat_models.ts [hunk 3]
```diff
@@ -257,67 +396,102 @@ export class ChatAnthropicMessages<
...
-            additional_kwargs: { ...data.delta },
-          }),
-          text: "",
-        });
-        if (data?.usage !== undefined) {
-          usageData.output_tokens += data.usage.output_tokens;
+    const requestOptions = this.invocationOptions(
+      {
+        ...params,
+        stream: false,
```
Comment (jacoblee93) [inline libs/langchain-anthropic/src/chat_models.ts:402]: Why reverse the order?

File: libs/langchain-anthropic/src/chat_models.ts [hunk 4]
```diff
@@ -463,31 +660,35 @@ export class ChatAnthropicMessages<
    * @returns A streaming request.
    */
   protected async createStreamWithRetry(
-    request: AnthropicStreamingMessageCreateParams & Kwargs
+    request: AnthropicStreamingMessageCreateParams & Kwargs,
+    options?: AnthropicRequestOptions
   ): Promise<Stream<AnthropicMessageStreamEvent>> {
     if (!this.streamingClient) {
-      const options = this.apiUrl ? { baseURL: this.apiUrl } : undefined;
+      const options_ = this.apiUrl ? { baseURL: this.apiUrl } : undefined;
```
Comment (jacoblee93) [inline libs/langchain-anthropic/src/chat_models.ts:667]: For the future: I despise this pattern

File: libs/langchain-anthropic/src/chat_models.ts [hunk 5]
```diff
@@ -502,19 +703,142 @@ export class ChatAnthropicMessages<
...
+      this.batchClient.messages.create(
+        {
+          ...request,
+          ...this.invocationKwargs,
+        } as AnthropicMessageCreateParams,
+        options
+      );
     return this.caller.callWithOptions(
-      { signal: options.signal },
+      { signal: options.signal ?? undefined },
```
Comment (jacoblee93) [inline libs/langchain-anthropic/src/chat_models.ts:714]: Why?

File: libs/langchain-anthropic/src/chat_models.ts [hunk 6]
```diff
@@ -502,19 +703,142 @@ export class ChatAnthropicMessages<
...
+    | Runnable<
+        BaseLanguageModelInput,
+        { raw: BaseMessage; parsed: RunOutput }
+      > {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const schema: z.ZodType<RunOutput> | Record<string, any> = outputSchema;
+    const name = config?.name;
+    const method = config?.method;
+    const includeRaw = config?.includeRaw;
+    if (method === "jsonMode") {
```
Comment (jacoblee93) [inline libs/langchain-anthropic/src/chat_models.ts:765]: We should make this type just `string` at some point

File: libs/langchain-anthropic/src/chat_models.ts (hunk 5 above)
Comment (bracesproul) [reply libs/langchain-anthropic/src/chat_models.ts:714]: signal in caller has to be signal or undefined, their sdk types it as signal, null or undefined and I'm using their types for this.

**Pull request merged**